plugins/libplugin-pay.c

```diff
@@ -386,6 +386,58 @@ void payment_start(struct payment *p)
 	payment_start_at_blockheight(p, INVALID_BLOCKHEIGHT);
 }
 
+static void channel_hint_to_json(const char *name, const struct channel_hint *hint, struct json_stream *dest)
+{
+	json_object_start(dest, name);
+	json_add_u32(dest, "timestamp", hint->timestamp);
+	json_add_short_channel_id_dir(dest, "scid", hint->scid);
+	json_add_amount_msat(dest, "capacity_msat", hint->estimated_capacity);
+	json_add_bool(dest, "enabled", hint->enabled);
+	json_object_end(dest);
+}
+
+/**
+ * Load a channel_hint from its JSON representation.
+ *
+ * @return The initialized `channel_hint` or `NULL` if we encountered a parsing
+ *         error.
+ */
+/*
+static struct channel_hint *channel_hint_from_json(const tal_t *ctx,
+						   const char *buffer,
+						   const jsmntok_t *toks)
+{
+	const char *ret;
+	struct channel_hint *hint = tal(ctx, struct channel_hint);
+	ret = json_scan(ctx, buffer, toks,
+			"{timestamp:%,scid:%,capacity_msat:%,enabled:%}",
+			JSON_SCAN(json_to_u32, &hint->timestamp),
+			JSON_SCAN(json_to_short_channel_id_dir, &hint->scid),
+			JSON_SCAN(json_to_msat, &hint->estimated_capacity),
+			JSON_SCAN(json_to_bool, &hint->enabled));
+
+	if (ret != NULL)
+		hint = tal_free(hint);
+	return hint;
+}
```

Comment:
I could have split this into a later commit, but it seemed like a good idea to have both the writing and reading part of the notification in one commit.